# DSCI 100 Project

## 1. Description

In [5]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [6]:
players <- read_csv("data/players.csv")
sessions <- read_csv("data/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


The players and sessions data tables were collected from Frank Wood's research study. In this study, players play "sessions" of minecraft and their data is collected. 

In [44]:
print(n = 10, tibble(players))
print(n = 10, tibble(sessions))

# A tibble: 196 × 7
   experience subscribe hashedEmail              played_hours name  gender   Age
   <chr>      <lgl>     <chr>                           <dbl> <chr> <chr>  <dbl>
 1 Pro        TRUE      f6daba428a5e19a3d475748…         30.3 Morg… Male       9
 2 Veteran    TRUE      f3c813577c458ba0dfef809…          3.8 Chri… Male      17
 3 Veteran    FALSE     b674dd7ee0d24096d1c0196…          0   Blake Male      17
 4 Amateur    TRUE      23fe711e0e3b77f1da7aa22…          0.7 Flora Female    21
 5 Regular    TRUE      7dc01f10bf20671ecfccdac…          0.1 Kylie Male      21
 6 Amateur    TRUE      f58aad5996a435f16b0284a…          0   Adri… Female    17
 7 Regular    TRUE      8e594b8953193b26f498db9…          0   Luna  Female    19
 8 Amateur    FALSE     1d2371d8a35c8831034b25b…          0   Emer… Male      21
 9 Amateur    TRUE      8b71f4d66a38389b7528bb3…          0.1 Nata… Male      17
10 Veteran    TRUE      bbe2d83de678f519c4b3daa…          0   Nyla  Female    22
# ℹ 186 

There are 196 players and 1535 sessions in the dataset. The players table has 7 variables: experience, subscribe, hashedEmail, played_hours, name, gender and Age. The sessions table has 5 variables: hashedEmail, start_time, end_time, original_start_time, original_end_time. Experience, subscribe and gender are categorical variables while played_hours, Age, start_time, end_time original_start_time, original_end_time are quantitative variables. One issue I can see with this table is that the columns are not consistently named. To fix this, Age should be age, hashedEmail should be hashed_email.

Summary Statistics

In [43]:
players_quantitative <- select(players, played_hours, Age)
map_df(players_quantitative, mean, na.rm = TRUE)
map_df(players_quantitative, median, na.rm = TRUE)
map_df(players_quantitative, sd, na.rm = TRUE)
map_df(players_quantitative, max, na.rm = TRUE)
map_df(players_quantitative, min, na.rm = TRUE)


played_hours Age     
1 5.845918     20.52062

played_hours Age
1 0.1          19

played_hours Age     
1 28.35734     6.174667

played_hours Age
1 223.1        50

played_hours Age
1 0            8

In [40]:
library(dplyr)

gender_counts <- players %>%
  select(gender) %>%
  count(gender)
gender_counts
subscribe_counts <- players %>%
  select(subscribe) %>%
  count(subscribe)
subscribe_counts

experience_counts <- players %>%
  select(experience) %>%
  count(experience)
experience_counts

gender            n  
1 Agender             2
2 Female             37
3 Male              124
4 Non-binary         15
5 Other               1
6 Prefer not to say  11
7 Two-Spirited        6

subscribe n  
1 FALSE      52
2  TRUE     144

experience n 
1 Amateur    63
2 Beginner   35
3 Pro        14
4 Regular    36
5 Veteran    48

73.5% players are subscribed. The mean number of played hours is 5.85 hours, the median is 0.1 hours, the maximum is 223 hours, the minimum is 0 hours, and the standard deviation is 28.35 hours. The mean age is 20.5 with a standard deviation of 6. The majority of players are male and expreience is mostly even with the exception of fewer pro players.

## 2. Questions

Can experience level predict subscription status?

## 3. Visualizations/Analysis

## 4. Methods/Plan

In [ ]:
I will use KNN 